src/spec/test/typing/TypeCheckingTest.groovy

```diff
@@ -44,6 +44,23 @@ class TypeCheckingTest extends StaticTypeCheckingTestCase {
         '''
     }
 
+    void testTypeCheckedEachWithIndex(){
+        //TODO: tags?
```

Comment:
Tests like this either belong here if they appear as part of the documentation in which case we'd also add the documentation and tags or we'd put them somewhere like:
https://github.com/apache/groovy/blob/master/src/test/groovy/transform/stc/DefaultGroovyMethodsSTCTest.groovy#L28-L40

Reply:
The current best fit for such doco would currently be core-syntax.adoc or core-semantics.adoc but I'll make a separate "working with arrays" subsection in the gdk doco since it would really belong there.

Reply:
I added the doco mentioned in previous comment, so if you want to add to the doco, you would update:
https://github.com/apache/groovy/blob/master/src/spec/test/gdk/WorkingWithArraysTest.groovy (probably `testArrayIteration`)
https://github.com/apache/groovy/blob/master/src/spec/doc/_working-with-arrays.adoc (if needed)
This wouldn't cover static compilation unless you added it explicitly, so moving to the aforementioned `DefaultGroovyMethodsSTCTest` might be just as good an option.

Reply:
Thank you very much for your help. I think I will add a simple test case here https://github.com/apache/groovy/blob/master/src/spec/test/gdk/WorkingWithArraysTest.groovy
that also works without my extension 
and one with stc into here https://github.com/apache/groovy/blob/master/src/test/groovy/transform/stc/DefaultGroovyMethodsSTCTest.groovy#L28-L40 that only works if we add the extension. Otherwise, I think it would be quite weird for the reader of the doco to see a method annotated with `@groovy.transform.TypeChecked`. 
One small note: I thought the doco file should be about arrays, but in the second paragraph you are talking about "Iterating on a list". Is this a maybe typo?